src/lib/variants.ts

```diff
@@ -209,6 +216,23 @@ export const withinMaxViews = (log: ViewLog, now: Date = new Date()): Filter =>
             minDaysBetweenViews: 0,
         };
 
+        // If we're looking at a special targeting test, we need some custom logic
+        // to see what variant a user is to see if they're within maxViews
+        if (test.name === epicTargetingHighlyEngagedTest.name) {
+            const variant = selectVariant(test, mvtId || 1);
+            const maxViews =
+                variant.name === 'control' ? CONTROL_MAX_VIEWS : HIGHLY_ENGAGED_VARIANT_MAX_VIEWS;
+
+            return !shouldThrottle(log, maxViews, test.name, now);
+        }
+        if (test.name === epicTargetingLessEngagedTest.name) {
+            const variant = selectVariant(test, mvtId || 1);
+            const maxViews =
+                variant.name === 'control' ? CONTROL_MAX_VIEWS : LESS_ENGAGED_VARIANT_MAX_VIEWS;
+
+            return !shouldThrottle(log, maxViews, test.name, now);
+        }
+
```

Comment:
Originally, I added an optional `maxViews` field to the `Variant` model but I decided to just hardcode that logic in here instead as it felt more explicit/clear to directly handle the special cases based on their test names.

Reply:
I was thinking we'd add a `maxViews` on the variant and test to see if they really should see it after the call to `selectVariant`. That would mean that they do not fall through to any others tests if they've reached max views for this test/variant.
I'm trying to think through the implications of either decision - I guess it doesn't make a difference if all other tests are disabled

Reply:
After speaking about this we decided to use a variant level setting and add some custom logic after the user has been assigned to test+variant to check if they're above their max views. We will treat tests that have variant level maxViews as always ask at this point in the code.